### PR TITLE
Temporal augmentation model training

### PR DESCRIPTION
**Pull request by ivanzvonkov**
- GP loss is not being computed correctly yet but gp_model can be commented out and autoregressive LSTM will run and can be used to make visualizations

### CODE REVIEW

File: src/utils/unrolled_lstm.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import math
```
Comment (joos2010kj) [inline src/utils/unrolled_lstm.py:1]: Just to confirm, this is the exact same copy as that from the forecaster repo?

File: train_ta.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
+from pathlib import Path
```
Comment (joos2010kj) [inline train_ta.py:1]: This is a mix of train.py from dvg and train.py from ai4eo, right?

File: train_ta.py [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+config["train_minibatch_size"] = len(train_loader)
+config["val_dataset_size"] = len(valid_loader.dataset)
+config["val_minibatch_size"] = len(valid_loader)
+
+print("\u2713 Data loaders initialized")
+
+# ----------------------------------------------------------------------------------------------------------------------
+# Initialize model
+# ----------------------------------------------------------------------------------------------------------------------
+class TemporalAugmentor(nn.Module):
```
Comment (joos2010kj) [inline train_ta.py]: This is the new class you created, correct?

**Review by joos2010kj — COMMENTED** [review]
Hi,

Can I ask a few questions first before I get into it?

File: train_ta.py (hunk 2 above)
Comment (ivanzvonkov) [reply train_ta.py:1]: Sure more or less

File: train_ta.py (hunk 3 above)
Comment (ivanzvonkov) [reply train_ta.py]: yes

File: src/utils/unrolled_lstm.py (hunk 1 above)
Comment (ivanzvonkov) [reply src/utils/unrolled_lstm.py:1]: yes

**Comment by ivanzvonkov** [comment]
@joos2010kj can I merge?

**Comment by joos2010kj** [comment]
Sorry I'm late -- yeah, please do.

I think I saw a couple errors while running this branch but I patched them up while running so I will share my PR for this.

**Comment by joos2010kj** [comment]
@ivanzvonkov
